cmake/bob_robotics.cmake

```diff
@@ -207,6 +188,31 @@ macro(BoB_module_custom)
     # Check we're on a supported platform
     check_platform(${PARSED_ARGS_PLATFORMS})
 
+    if(PARSED_ARGS_GAZEBO_PLUGINS)
+        # I'm sometimes getting linker errors when ld is linking against the
+        # static libs for BoB modules (because Gazebo plugins, as shared libs,
+        # are PIC, but the static libs seem not to be). So let's just compile
+        # everything as PIC.
+        if(GNU_TYPE_COMPILER)
+            add_definitions(-fPIC)
```

Comment:
Having hit thrilling issues of this sort with GeNN, I **think** you can use ``-fPIE`` here which may perform a tiny bit better

Reply:
It's good to know about this option but it seems not to work here :-(. SO says that it may not play nice with the ``-static`` option, which could be something to do with it: https://stackoverflow.com/questions/2463150/what-is-the-fpie-option-for-position-independent-executables-in-gcc-and-ld

Reply:
ahh my bad - that was probably the solution to another random problem :smile: